.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "lz4"]
+	path = lz4
+	url = https://github.com/lz4/lz4/
```

Comment:
Bundling of libraries is [difficult for packagers](https://docs.fedoraproject.org/en-US/fesco/Bundled_Software_policy/). I did not check other OSes but at least Fedora [does have lz4-static](https://koji.fedoraproject.org/koji/rpminfo?rpmID=33125970). Couldn't it just use the system `liblz4.a`? If not there is some build bug as after `make clean` and `make` I get:
```
  HOSTCC   compel/src/main-host.o
  HOSTCC   compel/arch/x86/src/lib/handle-elf-host.o
  HOSTCC   compel/src/lib/handle-elf-host.o
  CC       images/bpfmap-data.o
  HOSTCC   compel/src/lib/log-host.o
  HOSTLINK compel/compel-host-bin
  CC       images/apparmor.o
  CC       images/rseq.o
  LINK     images/built-in.o
make[1]: Leaving directory '/home/azul/azul/criu-git2/lz4/lib'
make[1]: Entering directory '/home/azul/azul/criu-git2/lz4/lib'
compiling dynamic library 1.9.4
make[1]: Leaving directory '/home/azul/azul/criu-git2/lz4/lib'
make[1]: Entering directory '/home/azul/azul/criu-git2/lz4/lib'
compiling dynamic library 1.9.4
creating versioned links
make[1]: Leaving directory '/home/azul/azul/criu-git2/lz4/lib'
make[1]: Entering directory '/home/azul/azul/criu-git2/lz4/programs'
creating versioned links
make[1]: Leaving directory '/home/azul/azul/criu-git2/lz4/lib'
make[1]: Entering directory '/home/azul/azul/criu-git2/lz4/programs'
make[1]: Leaving directory '/home/azul/azul/criu-git2/lz4/programs'
make[1]: Leaving directory '/home/azul/azul/criu-git2/lz4/programs'
cp: cannot create regular file './lz4': File exists
make: *** [Makefile:64: lz4] Error 1
make: Leaving directory '/home/azul/azul/criu-git2/lz4'
make: *** [Makefile:241: lz4/lib/liblz4.a] Error 2
make: *** Waiting for unfinished jobs....
make: Leaving directory '/home/azul/azul/criu-git2/lz4'
```

Reply:
The reason lz4 added as a submodule was the container we use to build CRIU may not have lz4, and the second, I'd like to use pretty fresh lz4 version which is good enough in compression and performance.

This is strange that you meet build failure with the lz4 submodule. I will check it.

Reply:
Any bundled library is a possible security hole needing separate updating. Recent enough library version is the business of OS vendors. But I do not mind, just my $0.02.